extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/3D.pulldown/Measure.pushbutton/script.py

```diff
@@ -0,0 +1,288 @@
+    if y_start.DistanceTo(y_end) > 0.001:
+        meshes.append(create_line_mesh(y_start, y_end, LINE_COLOR_Y))
+
+    # Z direction line (blue)
+    higher_point = point1 if point1.Z > point2.Z else point2
+    z_start = DB.XYZ(higher_point.X, higher_point.Y, lower_z)
+    z_end = DB.XYZ(higher_point.X, higher_point.Y, max(point1.Z, point2.Z))
+    if z_start.DistanceTo(z_end) > 0.001:
+        meshes.append(create_line_mesh(z_start, z_end, LINE_COLOR_Z))
+
+    # Diagonal line (yellow)
+    if point1.DistanceTo(point2) > 0.001:
+        meshes.append(create_line_mesh(point1, point2, LINE_COLOR_DIAG))
+
+    return meshes
+
+
+def delete_all_visual_aids():
+    """Delete all visual aids by clearing the DC3D server meshes."""
+    global dc3d_server
+
+    if dc3d_server:
+        dc3d_server.meshes = []
+        uidoc.RefreshActiveView()
+        logger.info("Cleared all visual aids")
+
+
+def perform_measurement():
+    """Perform the measurement workflow: pick points, create aids, update UI."""
+    global measurement_history, dc3d_server
+
+    try:
+        with forms.WarningBar(title="Pick first point"):
+            point1 = revit.pick_elementpoint(world=True)
+            if not point1:
+                return
+
+        with forms.WarningBar(title="Pick second point"):
+            point2 = revit.pick_elementpoint(world=True)
+            if not point2:
+                return
+
+        new_meshes = create_measurement_meshes(point1, point2)
+
+        # Add to existing meshes (don't replace)
+        if dc3d_server:
+            existing_meshes = dc3d_server.meshes if dc3d_server.meshes else []
+            dc3d_server.meshes = existing_meshes + new_meshes
+            uidoc.RefreshActiveView()
+
+        dx, dy, dz, diagonal = calculate_distances(point1, point2)
+
+        measure_window.point1_text.Text = "Point 1: {}".format(format_point(point1))
+        measure_window.point2_text.Text = "Point 2: {}".format(format_point(point2))
+        measure_window.dx_text.Text = "ΔX: {}".format(format_distance(dx))
+        measure_window.dy_text.Text = "ΔY: {}".format(format_distance(dy))
+        measure_window.dz_text.Text = "ΔZ: {}".format(format_distance(dz))
+        measure_window.diagonal_text.Text = "Diagonal: {}".format(
+            format_distance(diagonal)
+        )
+
+        # Add to history
+        history_entry = "Measurement {}:\n  P1: {}\n  P2: {}\n  ΔX: {}\n  ΔY: {}\n  ΔZ: {}\n  Diagonal: {}\n".format(
+            len(measurement_history) + 1,
+            format_point(point1),
+            format_point(point2),
+            format_distance(dx),
+            format_distance(dy),
+            format_distance(dz),
+            format_distance(diagonal),
+        )
+        measurement_history.append(history_entry)
+
+        # Keep only last MAX_HISTORY measurements
+        if len(measurement_history) > MAX_HISTORY:
+            measurement_history = measurement_history[-MAX_HISTORY:]
```

Comment:
instead of using a list, you can use a [deque](https://docs.python.org/2/library/collections.html#collections.deque) with `maxlen=MAX_HISTORY`.

You can then just `.append` without having to check the length of the collection and drop the `global` at the beginning of this function since you don't re-assign it.